pkgs/flox/default.nix

```diff
@@ -20,9 +20,9 @@ let
     if (FLOX_VERSION != null) then
       FLOX_VERSION
     else if !(self ? revCount || self ? shortRev) then
-      "${fileVersion}+git.dirty"
+      "${fileVersion}-dirty"
     else if !(self ? revCount) then
-      "${fileVersion}+git.${self.shortRev}"
+      "${fileVersion}-g${self.shortRev}"
```

Comment:
Maybe we should add some comments so we know where we depend on this and why we use this format? So we don't break it again in the future

Reply:
yes a _reason_ would be great, even though i know its probably https://github.com/semver/semver/issues/394